nservicebus/messaging/file-based-routing.md

```diff
@@ -0,0 +1,46 @@
+---
+title: File-based message routing
+summary: How to configure file-based message routing in NServiceBus
+tags:
+- routing
+- message
+- route
+- file
+- xml
+- dynamic
+related:
+- nservicebus/messaging/message-owner
+- nservicebus/messaging/routing
+---
+
+Before version 6, NServiceBus used fixed [message ownership mappings](/nservicebus/messaging/message-owner.md) in the standard .NET configuration file. The advantage was that operations personnel can adjust the routing should the topology change. This is especially important in case of bus-like transports (e.g. MSMQ) where machine name is part of the routing information. Version 6 introduced all-new [code-first API](/nservicebus/messaging/routing.md) that allows for better developer experience. In order to provide a better experience also for operations, we introduced file-based routing.
+
+## Configuration
+
+The following code
+
+snippet:Routing-FileBased-Config
```

Comment:
Placing a snippet inside a sentence will make for some fairly disjointed reading